test/src/test/java/hudson/tasks/BuildTriggerTest.java

```diff
@@ -309,13 +310,6 @@ public void downstreamProjectShouldObserveCompletedParent() throws Exception {
 
         final FreeStyleProject us = j.createFreeStyleProject();
         us.getPublishersList().add(new BuildTrigger("downstream", true));
-        us.getPublishersList().add(new TestNotifier() {
-            @Override
-            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-                Thread.sleep(5000); // Build does not complete just after the BuildTrigger
-                return true;
-            }
-        });
```

Comment:
> it happens a lot later

Well the downstream build should be _scheduled_ synchronously, but it can then take a moment for queue maintenance to happen. That is what the added `waitUntilNoActivity` is supposed to be handling.